server/src/main/java/org/apache/druid/discovery/DruidLeaderClient.java

```diff
@@ -133,6 +134,14 @@ public FullResponseHolder go(Request request) throws IOException, InterruptedExc
     return go(request, new FullResponseHandler(StandardCharsets.UTF_8));
   }
 
+  public <Intermediate, Final> ListenableFuture<Final> goStream(
```

Comment:
Why is this method needed? All callers immediately call `future.get()`. `go()` method is much better because it contains the retry mechanism considering changed leaders.

Reply:
Renamed to `goAsync` and added javadoc. I added it because the existing `go()` method return a `FullResponseHolder`, but the response handler needs to be passed to `JsonParserIterator` which expects a `Future<InputStream>`.

Reply:
Please add that this method doesn't do retrying on errors or handle leader changes occurred during communication to javadoc.